### PR TITLE
Support network alias

### PR DESCRIPTION
**Pull request by EdwardStudy**
This PR allows agent support to configure network alias. These changes have been worked well with SoftLayer.

Firstly, the bosh agent would get the manual network with 'alias' property from the agent settings file. Then, the agent would write to '/etc/network/interfaces' with the following important configuration:

```
// physical interface
auto eth0
iface eth0 inet static
    address 10.10.11.111
    network 10.10.11.0
    netmask 255.255.255.0

// virtual interface on eth0
auto eth0:0
iface eth0:0 inet static
    address 10.12.11.140
    network 10.12.11.128
    netmask 255.255.255.192
```

After ifup-ing these interfaces, the agent finished configuring VM networks with alias successfully.

@mattcui @maximilien @cppforlife  Please help to review. Thank you~

### CODE REVIEW

**Comment by dpb587-pivotal** [comment]
🔗 Pivotal Tracker, [Story #155716423](https://www.pivotaltracker.com/story/show/155716423)

**Comment by EdwardStudy** [comment]
Updated CLA.

Close and reopen it.

File: platform/net/ubuntu_net_manager_test.go [hunk 1]
```diff
@@ -904,6 +938,157 @@ iface ethstatic inet static
...
+    network 169.50.68.64
+    netmask 255.255.255.224
+    broadcast 169.50.68.95
+    gateway 169.50.68.65
+
+dns-nameservers 8.8.8.8 10.0.80.11 10.0.80.12`
+				Expect(networkConfig.StringContents()).To(Equal(expectedNetworkConfigurationForStaticAndDhcp))
+			})
+
+			It("fails", func() {
```
Comment (mikexuu) [inline platform/net/ubuntu_net_manager_test.go]: The test for the `fails` block is the same as the `succeeds` block above. Could you please update this test?

**Comment by EdwardStudy** [comment]
@mikexuu Sorry for late. I just updated test case and merged the latest code.

**Comment by mikexuu** [comment]
@EdwardStudy thanks for the PR, we just merged it.
Additionally, we decided to remove this [part of the PR](https://github.com/cloudfoundry/bosh-agent/pull/151/files#diff-d3c724c79695f4509b6d3f1e9fa227daR106). In the `SetupNetworking` function, a couple lines below we are already updating network interfaces / DHCP configurations with DNS information, so explicitly writing to resolv.conf seems like duplicate logic.
If you have any concerns about this change, please feel free to reopen this PR and let us know.

Thanks
@monkeyherder && @mikexuu

**Comment by EdwardStudy** [comment]
@mikexuu OK. Could you tell me which stable branch if I want to apply your changes to test?

**Comment by EdwardStudy** [comment]
@mikexuu Hi, Mike. I just checked [code](https://github.com/cloudfoundry/bosh-agent/blob/c74db775183f3e0e68254279e649b802bd4bf63f/platform/net/ubuntu_net_manager.go#L100-L103). SL networks in agent.json is not filled in ""

```

**Comment by mikexuu** [comment]
From what I could find, the `preconfigured` flag came from softlayer specific requests to manually write resolv.conf in this [commit](https://github.com/cloudfoundry/bosh-agent/commit/529366f78b1bd84b03fec2d3e94886bb32a98107)  and this related [PR](https://github.com/cloudfoundry/bosh-agent/pull/35).

**Comment by edwardstudy** [comment]
@mikexuu Hi, Mike. I testd failed bosh-agent and found bosh-agent should not simple returned when invoking SetupNetworking func in https://github.com/cloudfoundry/bosh-agent/blob/v2.91.3/platform/net/ubuntu_net_manager.go#L100-L103

Because bosh-agent should continue to configure `/etc/network/interfaces` file.

I will update another PR about this.

**Pull request merged**